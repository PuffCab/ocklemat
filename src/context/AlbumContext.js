import React, { createContext, useState } from "react";

export const AlbumContext = createContext();

export const AlbumContextProvider = ({ children }) => {
  const [albums, setAlbums] = useState();

  const [modal, setModal] = useState(false);
  const [clickedImg, setClickedImg] = useState("");
  const [slideIndex, setSlideIndex] = useState(0);

  const getData = async () => {
    try {
      const response = await fetch(
        "https://jsonplaceholder.typicode.com/albums/1/photos"
      );
      const obj = await response.json();
      // console.log(obj);
      setAlbums(obj);
    } catch (error) {
      console.log(error);
    }
  };

  return (
    <AlbumContext.Provider
      value={{
        albums,
        getData,
        modal,
        setModal,
        clickedImg,
        slideIndex,
        setSlideIndex,
      }}
    >
      {children}
    </AlbumContext.Provider>
  );
};
