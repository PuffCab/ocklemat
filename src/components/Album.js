import React, { useContext, useEffect, useState } from "react";
import { AlbumContext } from "../context/AlbumContext";
import AlbumImage from "./AlbumImage";

// import "./album.css";

function Album() {
  const { albums, getData } = useContext(AlbumContext);

  //   const [images, setimages] = useState([]);
  //   setimages(albums.url);

  console.log(albums);
  useEffect(() => {
    getData();
  }, []);

  return (
    <div className="gridContainer">
      {albums &&
        albums.map((item, index) => {
          return <AlbumImage key={index} image={item} />;
        })}
    </div>
  );
}

export default Album;
