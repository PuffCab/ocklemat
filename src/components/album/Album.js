import React, { useContext, useEffect } from "react";
import { AlbumContext } from "../../context/AlbumContext";
import AlbumImage from "../albumImage/AlbumImage";
import "./album.css";
import CloseIcon from "@material-ui/icons/Close";
import BtnModal from "./BtnModal";
import { Link, Route, useLocation, useHistory } from "react-router-dom";

function Album() {
  const { albums, getData, slideIndex, setSlideIndex } =
    useContext(AlbumContext);

  useEffect(() => {
    getData();
  }, []);

  ///////////
  //* Slider Controls
  ///////////

  const location = useLocation();
  const history = useHistory();
  const { state = {} } = location;
  const { modal } = state;

  const nextSlide = () => {
    if (slideIndex >= albums.length - 1) {
      location.pathname = `/album/${albums[0].id}`;
      setSlideIndex(0);
    } else {
      history?.push({
        pathname: `/album/${albums[slideIndex + 1].id}`,
        state: { modal: true },
      });
      setSlideIndex(slideIndex + 1);
    }
  };

  const prevSlide = () => {
    if (slideIndex === 0) {
      history?.push({
        pathname: `/album/${albums[albums.length - 1].id}`,
        state: { modal: true },
      });
      setSlideIndex(albums.length - 1);
    } else {
      history?.push({
        pathname: `/album/${albums[slideIndex - 1].id}`,
        state: { modal: true },
      });
      setSlideIndex(slideIndex - 1);
    }
  };

  return (
    <div className="Album_Wrapper">
      {albums && modal && (
        <Route
          path="/album/:id"
          children={({ match, history }) => {
            const id = +match?.params.id;
            const imageIndex =
              id && albums.findIndex((image) => image.id === id);
            setSlideIndex(imageIndex);

            const close = (e) => {
              e.stopPropagation();
              history.push("/");
            };

            return (
              <div className="modal open">
                <div>
                  <img
                    className="modal-image"
                    src={albums[slideIndex].url}
                    alt={albums[slideIndex].title}
                  />

                  <CloseIcon className="modal-close-btn" onClick={close}>
                    close modal
                  </CloseIcon>

                  <BtnModal moveSlide={nextSlide} direction={"next"} />
                  <BtnModal moveSlide={prevSlide} direction={"prev"} />
                </div>
              </div>
            );
          }}
        />
      )}

      <div className="Album_GridContainer">
        {albums &&
          albums.map((item, index) => {
            return (
              <Link
                to={{ pathname: `/album/${item.id}`, state: { modal: true } }}
              >
                {" "}
                <AlbumImage
                  className="Album_gridImage"
                  key={index}
                  image={item}
                />
              </Link>
            );
          })}
      </div>
    </div>
  );
}

export default Album;
