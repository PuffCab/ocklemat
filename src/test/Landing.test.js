import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Album from "../components/album/Album";
import { AlbumContextProvider } from "../context/AlbumContext";

it("Landing component should render without crashing", () => {
  shallow(
    <AlbumContextProvider>
      <Album />
    </AlbumContextProvider>
  );
});

it("landing component snapshot with Enzyme", () => {
  const wrapper = shallow(
    <AlbumContextProvider>
      <Album />
    </AlbumContextProvider>
  );
  expect(toJson(wrapper)).toMatchSnapshot();
});
