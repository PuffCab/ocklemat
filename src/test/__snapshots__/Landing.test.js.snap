// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`landing component snapshot with Enzyme 1`] = `
<ContextProvider
  value={
    Object {
      "albums": undefined,
      "clickedImg": "",
      "getData": [Function],
      "modal": false,
      "setModal": [Function],
      "setSlideIndex": [Function],
      "slideIndex": 0,
    }
  }
>
  <Album />
</ContextProvider>
`;
